indi-asi/asi_ccd.cpp

```diff
@@ -129,12 +147,184 @@ static class Loader
+
+void ASICCD::loadNicknames()
+{
+    const std::string filename = GetHomeDirectory() + NICKNAME_FILE;
+    mNicknames.clear();
+
+    LilXML *xmlHandle = newLilXML();
+    XMLEle *rootXmlNode = nullptr;
+    char errorMessage[512] = {0};
+    FILE *file = fopen(filename.c_str(), "r");
+    if (file)
+    {
+        rootXmlNode = readXMLFile(file, xmlHandle, errorMessage);
+        fclose(file);
+    }
+    delLilXML(xmlHandle);
+
+    if (rootXmlNode == nullptr)
+        return;
+
+    XMLEle *currentXmlNode = nextXMLEle(rootXmlNode, 1);
+    while (currentXmlNode)
+    {
+        const char *id = findXMLAttValu(currentXmlNode, ATTRIBUTE);
+        if (id != nullptr)
+        {
+            std::string name = pcdataXMLEle(currentXmlNode);
+            if (!name.empty())
+                trim(name);
+            if (!name.empty())
+                mNicknames[id] = name;
+        }
+        currentXmlNode = nextXMLEle(rootXmlNode, 0);
+    }
+
+    delXMLEle(rootXmlNode);
+}
+
+void ASICCD::saveNicknames()
+{
+    const std::string filename = GetHomeDirectory() + NICKNAME_FILE;
+    XMLEle *rootXmlNode = nullptr;
+    XMLEle *oneElement = nullptr;
+
+    FILE *file = fopen(filename.c_str(), "w");
+
+    rootXmlNode = addXMLEle(nullptr, ROOTNODE);
+
+    for (const auto &kv : mNicknames)
+    {
+        oneElement = addXMLEle(rootXmlNode, ENTRYNODE);
+        addXMLAtt(oneElement, ATTRIBUTE, kv.first.c_str());
+        editXMLEle(oneElement, kv.second.c_str());
+    }
+
+    prXMLEle(file, rootXmlNode, 0);
+    fclose(file);
+    delXMLEle(rootXmlNode);
+}
+
+
+bool ASICCD::ISNewText(const char *dev, const char *name, char *texts[], char *names[], int n)
+{
+    if (dev != nullptr && !strcmp(dev, getDeviceName()))
+    {
+        if (NicknameTP.isNameMatch(name))
+        {
+            IUUpdateText(&NicknameTP, texts, names, n);
+            NicknameTP.setState(IPS_OK);
+            IDSetText(&NicknameTP, nullptr);
+            if (!mSerialNumber.empty())
+            {
+                loadNicknames();  // another camera may have updated its nickname.
+                std::string newNickname = texts[0];
+                trim(newNickname);
+                if (newNickname.empty())
+                {
+                    mNicknames.erase(mSerialNumber);
+                    LOGF_INFO("Nickname for %s removed.", mSerialNumber.c_str());
+                }
+                else
+                {
+                    mNicknames[mSerialNumber] = newNickname;
+                    LOGF_INFO("Nickname for %s changed to %s.", mSerialNumber.c_str(), newNickname.c_str());
+                }
+                saveNicknames();
+                LOG_INFO("The driver must now be restarted for this change to take effect.");
+            }
+            else
+            {
+                LOG_INFO("Can't apply nickname change--serial number not known.");
+            }
+            NicknameTP.apply();
+        }
+        return true;
```

Comment:
This return and the next should be inside the brace by one nesting level, now other text edits to the device like observer name are ignored.

Reply:
@jpaana I moved the return true up one level. I think the one at the bottom is ok where it is. Please double check, and if not, please let me know and I'll change it too.

Reply:
Thanks, works now. I've usually used an early return false at the top of the function if the device is null or doesn't match, as calling the base class function is also unnecessary if the call isn't for our device in the first place, it just does the same check again and returns false.

Reply:
I guess I'll leave as is and resolve.